src/filter.ts

```diff
@@ -0,0 +1,53 @@
+import * as LSIF from 'lsif-protocol';
+
+export interface IFilter {
+    id: string[];
```

Comment:
Feel free to ignore this, but I _feel_ as though these should all have `string[] | undefined` if they don't need to be a value.

Reply:
You are right for the "regex" property, but all of the others should be an empty array if nothing is specified (and not undefined).